### PR TITLE
Run 'pyupgrade --py36-plus' on all *.py files

### PR DESCRIPTION
**Pull request by cmarqu**
Depends on #2422.

### CODE REVIEW

**Comment by ktbarrett** [comment]
Oh no the 3.5 test is failing!

**Review by ktbarrett — COMMENTED** [review]
Any idea why it uses `.format()` in some cases and f-strings in others?

**Comment by marlonjames** [comment]
@ktbarrett Maybe it does `%s` -> `.format()` and `.format()` -> `f"{}"`, so those need a second pass?

File: cocotb/decorators.py [hunk 1]
```diff
@@ -333,13 +333,13 @@ def abort(self, exc):
...
+            self.log.debug(f"outcome forced to {outcome}")
         self._outcome = outcome
         cocotb.scheduler._unschedule(self)
 
     def sort_name(self):
         if self.stage is None:
-            return "%s.%s" % (self.module, self.funcname)
+            return f"{self.module}.{self.funcname}"
         else:
             return "%s.%d.%s" % (self.module, self.stage, self.funcname)
```
Comment (marlonjames) [inline cocotb/decorators.py:344]: It missed this.

File: cocotb/regression.py [hunk 2]
```diff
@@ -188,7 +188,7 @@ def _discover_tests() -> Iterable[Test]:
                         test = getattr(module, test_name)
                     except AttributeError:
                         _logger.error("Requested test %s wasn't found in module %s", test_name, module_name)
```
Comment (marlonjames) [inline cocotb/regression.py:190]: Looks like it doesn't find this type of logging that uses `%s` but not the `%` operator.

Comment (ktbarrett) [reply cocotb/regression.py:190]: Well that is used by the logger so that is expected.

**Comment by ktbarrett** [comment]
@cmarqu I just ran the command on your branch and it fixed even more stuff. Are you using the most up to date version?

**Comment by ktbarrett** [comment]
From [their docs](https://github.com/asottile/pyupgrade#f-strings).

> note: pyupgrade is intentionally timid and will not create an f-string if it would make the expression longer or if the substitution parameters are anything but simple names or dotted names (as this can decrease readability).

File: cocotb/decorators.py (hunk 1 above)
Comment (cmarqu) [reply cocotb/decorators.py:344]: This is indeed interesting.

File: cocotb/regression.py (hunk 2 above)
Comment (cmarqu) [reply cocotb/regression.py:190]: Yes, it appears to leave loggers lazy evaluation alone.

**Comment by cmarqu** [comment]
> From [their docs](https://github.com/asottile/pyupgrade#f-strings).
> 
> > note: pyupgrade is intentionally timid and will not create an f-string if it would make the expression longer or if the substitution parameters are anything but simple names or dotted names (as this can decrease readability).
> 
> I think we are going to have to wipe those manually, or find another tool.

This fork here [supposedly](https://github.com/getsentry/sentry/pull/23506) is more aggressive: https://github.com/joshuarli/pyupgrade/tree/joshuarli/individual-fixers

**Comment by ktbarrett** [comment]
Needs rebase.

File: tests/test_cases/test_array/test_array.py [hunk 3]
```diff
@@ -62,7 +62,7 @@ async def test_read_write(dut):
         _check_int(tlog, dut.param_char, ord('p'))
         _check_str(tlog, dut.param_str , b"ARRAYMOD")
 
-        if not cocotb.SIM_NAME.lower().startswith(("riviera")):
+        if not cocotb.SIM_NAME.lower().startswith("riviera"):
```
Comment (imphil) [inline tests/test_cases/test_array/test_array.py:65]: Slightly unsure about that one -- we are trying to always pass a set here. But no strong feelings either way.

**Review by imphil — COMMENTED** [review]
This PR contains a couple changes where we went from `func((stuff))` to `func(stuff)`. I didn't double-check them, but it feels like the current code was actually intended that way. Could you have a look?

File: tests/test_cases/test_array/test_array.py (hunk 3 above)
Comment (ktbarrett) [reply tests/test_cases/test_array/test_array.py:65]: `(value)` is equivalent to `value`. What should have been here was `(value,)` which is a tuple literal. But it is correct for now, `startswith` can take a single string or a tuple/list of strings.

Comment (cmarqu) [reply tests/test_cases/test_array/test_array.py:65]: Indeed, I think this was a mistaken way of preparing for future extensions of that set.

**Review by ktbarrett — APPROVED** [review]
It's an automated process and the regression pass. I seriously doubt it go anything wrong. And it looks like it at least sort of helped remove out of date styles. It did totally miss the attribute annotations in handle.py, and a lot of the Python 2 string formatting. But we can do that in a follow on.

**Comment by ktbarrett** [comment]
Feel free to merge whenever.

**Pull request merged**